src/swarm/neo/protocol/socket/MessageReceiver.d

```diff
@@ -90,6 +90,23 @@ private class MessageReceiverBase
 
     protected size_t buffer_content_end = 0;
 
+    /***************************************************************************
+
+        The maximum number of subsequent `read` calls without an I/O-wait call.
+
+    ***************************************************************************/
+
+    private const uint read_iowait_max = 10;
+
+    /***************************************************************************
+
+        Counts the maximum number of subsequent `read` calls without an I/O-wait
+        call.
+
+    ***************************************************************************/
+
+    private uint read_iowait_count;
```

Comment:
Yes, but a local counter variable must be stored in the loop. The loop is not in `ensureMinimumAmountOfBytesInBuffer` but in `ConnectionBase.ReceiveLoop.fiberMethod`, the indirect caller. A local counter in `ensureMinimumAmountOfBytesInBuffer` would be 0 on each repeated call.